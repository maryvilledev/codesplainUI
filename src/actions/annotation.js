export const OPEN_ANNOTATION_PANEL = 'OPEN_ANNOTATION_PANEL';
export const CLOSE_ANNOTATION_PANEL = 'CLOSE_ANNOTATION_PANEL';

export const openAnnotationPanel = (snippetInformation) => ({
  type: OPEN_ANNOTATION_PANEL,
  payload: snippetInformation,
});

export const closeAnnotationPanel = () => ({
  type: CLOSE_ANNOTATION_PANEL,
});
