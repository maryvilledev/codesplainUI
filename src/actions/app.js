import axios from 'axios';

export const EDIT_ANNOTATION = 'EDIT_ANNOTATION';
export const PARSE_SNIPPET = 'PARSE_SNIPPET';
export const RESTORE_STATE = 'RESTORE_STATE';
export const SAVE_ANNOTATION = 'SAVE_ANNOTATION';
export const SET_AST = 'SET_AST';
export const SET_RULE_FILTERS = 'SET_RULE_FILTERS';
export const SET_SNIPPET_CONTENTS = 'SET_SNIPPET_CONTENTS';
export const SET_SNIPPET_LANGUAGE = 'SET_SNIPPET_LANGUAGE';
export const SET_SNIPPET_TITLE = 'SET_SNIPPET_TITLE';
export const TOGGLE_EDITING_STATE = 'TOGGLE_EDITING_STATE';
export const SAVE_STATE = 'SAVE_STATE';
export const SAVE_STATE_STARTED = 'SAVE_STATE_STARTED';
export const SAVE_STATE_SUCCEEDED = 'SAVE_STATE_SUCCEEDED';
export const SAVE_STATE_FAILED = 'SAVE_STATE_FAILED';


export const setSnippetContents = (snippet) => ({
  type: SET_SNIPPET_CONTENTS,
  payload: snippet,
});

export const setRuleFilters = (filters) => ({
  type: SET_RULE_FILTERS,
  payload: filters,
});

export const setAST = (AST) => ({
  type: SET_AST,
  payload: AST,
});

export const toggleEditState = () => ({
  type: TOGGLE_EDITING_STATE,
});

export const saveAnnotation = (annotationData) => ({
  type: SAVE_ANNOTATION,
  payload: annotationData,
});

export const restoreState = (savedState) => ({
  type: RESTORE_STATE,
  payload: savedState,
});

export const setSnippetTitle = (snippetTitle) => ({
  type: SET_SNIPPET_TITLE,
  payload: snippetTitle
})

export const setSnippetLanguage = (snippetLanguage) => ({
  type: SET_SNIPPET_LANGUAGE,
  payload: snippetLanguage
});

export const parseSnippet = (snippet) => ({
  type: PARSE_SNIPPET,
  meta: {
    WebWorker: true,
  },
  payload: {
    snippet,
  },
});

export const saveStateStarted = () => ({
  type: SAVE_STATE_STARTED,
});

export const saveStateSucceeded = () => ({
  type: SAVE_STATE_SUCCEEDED,
});

export const saveStateFailed = () => ({
  type: SAVE_STATE_FAILED,
});

export const saveState = (id) => {
  return (dispatch, getState) => {
    // If the snippet has not been saved, then saving an annotation shouldn't
    // trigger a POST request to save the snippet (and its annotations)
    if (id === undefined) {
      // Return a Promise object to make this action "thenable"
      return Promise.resolve();
    }
    dispatch(saveStateStarted());
    // Get the app state and stringify it
    const stateString = JSON.stringify(getState().app);

    // TODO: Dispatch action to show the snackbar with a relevant message based on the results
    // Return the results of the POST request
    return axios.post(`api/snippets/${id}`, { json: stateString, })
      .then((res) => {
        dispatch(saveStateSucceeded());
      }, (err) => {
        dispatch(saveStateFailed());
      });
  };
};
