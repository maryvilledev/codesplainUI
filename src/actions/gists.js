export const SET_GISTS = 'SET_GISTS';

export const setGists = gists => ({
  type: SET_GISTS,
  payload: gists,
});
