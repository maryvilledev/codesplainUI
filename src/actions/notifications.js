export const ADD_NOTIFICATION = 'ADD_NOTIFICATION';
export const CLOSE_NOTIFICATION = 'CLOSE_NOTIFICATION';

export const addNotification = notification => ({
  type: ADD_NOTIFICATION,
  payload: notification,
});

export const closeNotification = () => ({
  type: CLOSE_NOTIFICATION,
});
