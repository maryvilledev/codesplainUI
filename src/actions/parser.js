export const LOAD_PARSER = 'LOAD_PARSER';

export const loadParser = language => ({
  type: LOAD_PARSER,
  meta: {
    WebWorker: true,
  },
  payload: language,
});
