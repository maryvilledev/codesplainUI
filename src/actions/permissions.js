export const SET_PERMISSIONS = 'SET_PERMISSIONS';

export const setPermissions = permissions => ({
  type: SET_PERMISSIONS,
  payload: permissions,
});
