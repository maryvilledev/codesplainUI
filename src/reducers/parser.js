import * as actions from '../actions/parser';

const initialState = {
  language: '',
};

const parser = (state = initialState, action) => {
  switch (action.type) {
  case actions.LOAD_PARSER: {
    if (action.meta) {
        // Missed the middleware somehow
      return state;
    }
    const language = action.payload;
    return { ...state, language };
  }
  default: {
    return state;
  }
  }
};

export default parser;
