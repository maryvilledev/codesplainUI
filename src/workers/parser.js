import { LOAD_PARSER } from '../actions/parser';
import { PARSE_SNIPPET } from '../actions/app';
import onError from '../util/parser-error-logger.js';
import { getRuleCount, rules } from '../util/rules.js';
import { makeParserEndpointUrl } from '../util/requests.js';

const API_URL = process.env.REACT_APP_API_URL;

let parser = null;
const parserCache = {};

self.onmessage = ({ data: action }) => {
  switch (action.type) {
  case PARSE_SNIPPET: {
    if (!parser) break;
    const { snippet } = action.payload;
    const AST = parser(snippet, onError);
    const ruleCounts = {}; getRuleCount(AST, ruleCounts);
    self.postMessage({
      type: action.type,
      payload: {
        AST,
        ruleCounts,
      },
    });
    break;
  }
  case LOAD_PARSER: {
    const language = action.payload;
    // Check to see if we already have the parser loaded.
    if (language in parserCache) {
      parser = parserCache[language];
    } else {
      // Load and execute script from the URL
      self.importScripts(makeParserEndpointUrl(language));
      // Script exports the parser as a global var, set the local ref to point
      // at it
      parser = CodesplainParser;
      // Add the parser to the cache so we don't have to load it again
      parserCache[language] = parser;
    }
    self.postMessage({
      type: action.type,
      payload: action.payload,
    });
    break;
  }
  default: {
    console.log(action.type);
  }
  }
};
