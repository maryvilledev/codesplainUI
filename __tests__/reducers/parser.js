import * as actions from '../../src/actions/parser';
import reducer from '../../src/reducers/parser';

describe('Reducer: Parser', () => {
  it('should have initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle LOAD_PARSER', () => {
    const language = 'squanchy';
    const action = {
      type: actions.LOAD_PARSER,
      payload: language,
    };
    const expected = {
      language,
    };
    expect(reducer(undefined, action)).toEqual(expect.objectContaining(expected));
  });
});
