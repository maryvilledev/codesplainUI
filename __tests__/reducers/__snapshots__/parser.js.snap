exports[`Reducer: Parser should have initial state 1`] = `
Object {
  "language": "",
}
`;
